scripts/playwright-setup

```diff
@@ -0,0 +1,150 @@
+#!/usr/bin/env bash
+# Build script to handle configuring Playwright for running integration tests,
+# both as a "pretest" hook in `package.json` and via manual invocation in CI.
+#
+# By default this will run `pnpm exec playwright install` because that's the sane thing to do,
+# and will work on most normal computers, such as macOS. The bulk of the logic is to handle
+# special cases like non-Playwright-supported Linux distributes, such as Fedora Workstation.
+# Additionally, the special-case logic *only* applies if the script is being run from within
+# a nix devShell. So if you're not using nix, no problem, you don't have to. But if you are, this script
+# will wire up integration testing via Playwright by committing some grave sins in symlink form.
+#
+# Specifically, it will inspect the contents of `~/.cache/ms-playwright`, which is a directory
+# that Playwright will *always* check for browsers. If it finds pre-existing Playwright data,
+# it'll preserve it and emit a warning about tests maybe not working right. If no prior Playwright
+# data exists, it'll configure the symlinks, so that when playwright looks for a headless chromium,
+# it'll find the nix-installed version that's present in the build store.
+set -euo pipefail
+
+# This is a hardcoded value that matches exactly the path Playwright
+# will check within its browser cache dir. This value must be updated
+# whenever the `playwright` npm dependency is changed. If it breaks, mention @conorsch.
+_PLAYWRIGHT_CHROMIUM_HEADLESS_VERSION="1169"
+
+
+# If we're not in a nix shell, then just do the simple thing:
+# defer to playwright docs on default setup strategy.
+if [[ -z "${IN_NIX_SHELL:-}" ]] ; then
+  >&2 echo "Installing playwright..."
+  pnpm exec playwright install
+  exit 0
+fi
```

Comment:
Careful attention was paid to ensure that _only_ nix devshells get to grapple with the ratking of if statements below.